File: mobile/src/main/java/com/windscribe/mobile/connectionsettings/ConnectionSettingsView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.connectionsettings;

import java.util.List;

public interface ConnectionSettingsView {

    void getLocationPermission(int requestCode);

    void gotoSplitTunnelingSettings();

    void openGpsSpoofSettings();

    void packetSizeDetectionProgress(boolean progress);

    void setAutoStartOnBootToggle(int toggleDrawable);

    void setGpsSpoofingToggle(int toggleDrawable);

    void setKeepAlive(String keepAlive);

    void setLanBypassToggle(int toggleDrawable);

    void setPacketSize(String size);

    void setSplitTunnelText(String onOff, Integer color);

    void setupConnectionModeAdapter(String savedValue, String[] connectionModes);

    void setupPacketSizeModeAdapter(String savedValue, String[] types);

    void setKeepAliveModeAdapter(String savedValue, String[] types);

    void setupPortMapAdapter(String port, List<String> portMap);

    void setupProtocolAdapter(String protocol, String[] mProtocols);

    void showGpsSpoofing();

    void showToast(String toastString);

    void setDecoyTrafficToggle(int toggleDrawable);

    void showExtraDataUseWarning();

    void setupFakeTrafficVolumeAdapter(String selectedValue, String[] values);

    void setPotentialTrafficUse(String value);

    void showAutoStartOnBoot();

    void setKeepAliveContainerVisibility(boolean isAutoKeepAlive);
}

File: mobile/src/main/java/com/windscribe/mobile/connectionsettings/ConnectionSettingsView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.connectionsettings

import android.os.Build
import androidx.annotation.RequiresApi

interface ConnectionSettingsView {
    fun gotoSplitTunnelingSettings()
    fun goToNetworkSecurity()
    fun openGpsSpoofSettings()
    fun packetSizeDetectionProgress(progress: Boolean)
    fun setAutoStartOnBootToggle(toggleDrawable: Int)
    fun setGpsSpoofingToggle(toggleDrawable: Int)
    fun setKeepAlive(keepAlive: String)
    fun setLanBypassToggle(toggleDrawable: Int)
    fun setPacketSize(size: String)
    fun setSplitTunnelText(onOff: String, color: Int)
    fun setupConnectionModeAdapter(savedValue: String, connectionModes: Array<String>)
    fun setupPacketSizeModeAdapter(savedValue: String, types: Array<String>)
    fun setKeepAliveModeAdapter(savedValue: String, types: Array<String>)
    fun setupPortMapAdapter(port: String, portMap: List<String>)
    fun setupProtocolAdapter(protocol: String, protocols: Array<String>)
    fun showGpsSpoofing()
    fun showToast(toastString: String)
    fun setDecoyTrafficToggle(toggleDrawable: Int)
    fun showExtraDataUseWarning()
    fun setupFakeTrafficVolumeAdapter(selectedValue: String, values: Array<String>)
    fun setPotentialTrafficUse(value: String)
    fun showAutoStartOnBoot()
    fun setKeepAliveContainerVisibility(isAutoKeepAlive: Boolean)
    fun setAntiCensorshipToggle(toggleDrawable: Int)
    fun setAutoConnectToggle(toggleDrawable: Int)
    fun setupDNSModeAdapter(savedValue: String, dnsModes: Array<String>)
    fun setCustomDnsAddress(dnsAddress: String)
    fun setPowerWhitelistToggle(toggleDrawable: Int)
    @RequiresApi(Build.VERSION_CODES.M)
    fun launchBatteryOptimizationActivity()
}